packages/core/src/client.ts

```diff
@@ -1795,15 +1795,17 @@ export class MedplumClient extends EventTarget {
    * @param filename - Optional filename for the binary.
    * @param contentType - Content type for the binary.
    * @param onProgress - Optional callback for progress events.
+   * @param signal - Optional AbortSignal to abort the request.
    * @returns The result of the create operation.
    */
   async createAttachment(
     data: BinarySource,
     filename: string | undefined,
     contentType: string,
-    onProgress?: (e: ProgressEvent) => void
+    onProgress?: (e: ProgressEvent) => void,
+    signal?: AbortSignal
```

Comment:
How would you feel about changing this param to `options: RequestInit = {}`?

My thinking:
1. It aligns with other methods in the client
2. It futureproofs the method for predictable future requests for setting HTTP headers, etc

There would be separate work to actually use the values of `RequestInit` with `XMLHttpRequest`, but I'm ok punting on that for now.

Reply:
> There would be separate work to actually use the values of RequestInit with XMLHttpRequest

Yeah this is tough because if someone is relying on an option of `RequestInit` and they then add an `onProgress` callback, they may unexpectedly change the behavior. If this were very explicitly documented somewhere, I could see getting behind it more.

Reply:
yup sounds good - added here https://github.com/medplum/medplum/pull/3420/commits/d8fb4c87a63455b1ce3d378b5981b57e61fee3dc